ledger/accountdb.go

```diff
@@ -1968,29 +1985,77 @@ func (iterator *orderedAccountsIter) Next(ctx context.Context) (acct []accountAd
 
 			copy(addr[:], addrbuf)
 
-			var accountData basics.AccountData
+			var accountData baseAccountData
 			err = protocol.Decode(buf, &accountData)
 			if err != nil {
 				iterator.Close(ctx)
 				return
 			}
-			hash := accountHashBuilder(addr, accountData, buf)
+			hash := accountHashBuilderV6(addr, &accountData, buf)
 			_, err = iterator.insertStmt.ExecContext(ctx, addrbuf, hash)
 			if err != nil {
 				iterator.Close(ctx)
 				return
 			}
 
+			resourcesEntriesCount := accountData.TotalAppParams + accountData.TotalAppLocalStates + accountData.TotalAssetParams + accountData.TotalAssets
+			for ; resourcesEntriesCount > 0; resourcesEntriesCount-- {
+				if !iterator.resourcesRows.Next() {
+					iterator.Close(ctx)
+					err = errors.New("missing entries in resource table")
```

Comment:
It would be nice for errors.New() to only occur in a var declaration and wrapped e.g. `fmt.Errorf("missing entry for %s in resource table: %w", addr.String(), errMismatchedAccountResources)` or similar. That would also provide the benefit of identifying the offending address with the wrong resources.

That reminds me that I was hoping to enable the go-err113 linter which would complain "do not define dynamic errors, use wrapped static errors instead" for cases like this.